### ISSUE TITLE
formatter/lsp: `overrides` in `.oxfmtrc.json` not applied when formatting via VS Code extension

### ISSUE THREAD

**Issue by hossam-magdy** [oxc-project/oxc#21385]
## Summary

The Oxfmt LSP server does not apply `overrides` from `.oxfmtrc.json` when formatting via the VS Code extension. The CLI correctly applies overrides. This causes the extension to format files at the root `printWidth` (e.g. 80), ignoring per-file overrides (e.g. `printWidth: 120`), producing a massive diff on save.

### Project structure

```
monorepo/                              ← VS Code workspace root
├── packages/
│   └── my-package/
│       ├── .oxfmtrc.json
│       └── src/
│           └── deep/
│               └── LargeFile.tsx      ← file being formatted
```

### `.oxfmtrc.json` (in `packages/my-package/`)

```json
{
  "printWidth": 80,
  "overrides": [
    {
      "files": ["src/deep/LargeFile.tsx"],
      "options": { "printWidth": 120 }
    }
  ]
}
```

### Steps

1. Open the monorepo root in VS Code / Cursor
2. Open `packages/my-package/src/deep/LargeFile.tsx`
3. Format via extension (either `source.format.oxc` code action on save, or `editor.defaultFormatter: "oxc.oxc-vscode"`)
4. Observe: file is formatted at `printWidth: 80` (the root default), ignoring the override

### Expected

File should be formatted at `printWidth: 120` per the override, matching CLI behavior.

### CLI works correctly

```sh
cd packages/my-package
npx oxfmt --check .
# ✅ All matched files use the correct format.
```

The CLI (run from the package directory) correctly resolves the config, applies the override for `LargeFile.tsx`, and reports no formatting issues.

## Environment

- `oxfmt` CLI: `0.42.0` (installed in workspace `node_modules`)
- `oxc-vscode` extension: `1.52.0`
- Editor: Cursor (VS Code fork)
- OS: macOS 15 (Apple Silicon)

## Analysis

The extension sends `fmt.configPath` to the LSP server via `toOxfmtConfig()`. When no explicit path is set, the server needs to discover `.oxfmtrc.json` by walking up from the file being formatted. It appears the LSP server either:

1. Does not walk up directories to find the nearest `.oxfmtrc.json` (only checks workspace root), or
2. Finds the config but does not apply `overrides` when resolving options for a specific file

The CLI binary handles both correctly.

**Comment by hossam-magdy** [comment]
[EDITED]: -> moved to separate issue https://github.com/oxc-project/oxc/issues/21398

**Comment by camc314** [comment]
@hossam-magdy can you make a separate issue for the linter problem? using the same issue for both is going to be confusing - thanks

**Comment by hossam-magdy** [comment]
@camc314 Created separate issue https://github.com/oxc-project/oxc/issues/21398

**Comment by Sysix** [comment]
This is not a problem with overrides. LSP and CLI only looks at the working directory. For VS Code this is your workspace root.
https://github.com/oxc-project/oxc/pull/21081 will fix this problem for you.

**Comment by leaysgur** [comment]
> This is not a problem with overrides. LSP and CLI only looks at the working directory.

Right.

In that case, I'll close this issue as a duplicate.
